DataFormats/CaloRecHit/interface/CaloRecHit.h

```diff
@@ -4,27 +4,63 @@
 #include "DataFormats/DetId/interface/DetId.h"
 #include <ostream>
 
+#ifdef __CUDA_ARCH__
+  __constant__
+#else
+  constexpr
+#endif
+uint32_t masks[] = {
+  0x00000000u,0x00000001u,0x00000003u,0x00000007u,0x0000000fu,0x0000001fu,
+  0x0000003fu,0x0000007fu,0x000000ffu,0x000001ffu,0x000003ffu,0x000007ffu,
+  0x00000fffu,0x00001fffu,0x00003fffu,0x00007fffu,0x0000ffffu,0x0001ffffu,
+  0x0003ffffu,0x0007ffffu,0x000fffffu,0x001fffffu,0x003fffffu,0x007fffffu,
+  0x00ffffffu,0x01ffffffu,0x03ffffffu,0x07ffffffu,0x0fffffffu,0x1fffffffu,
+  0x3fffffffu,0x7fffffffu,0xffffffffu};
 
 /** \class CaloRecHit
  * 
  *\author J. Mans - Minnesota
  */
 class CaloRecHit {
 public:
-  CaloRecHit(); // for persistence
-  explicit CaloRecHit(const DetId& id, float energy, float time, uint32_t flags = 0, uint32_t aux=0);
+  constexpr CaloRecHit() : energy_(0), time_(0), flags_(0), aux_(0) {}
+  constexpr explicit CaloRecHit(const DetId& id, float energy, float time, 
+                                uint32_t flags = 0, uint32_t aux=0)
+	: id_(id),energy_(energy), time_(time), flags_(flags), aux_(aux) {}
 
-  float energy() const { return energy_; }
-  void setEnergy(float energy) { energy_=energy; }
-  float time() const { return time_; }
-  void setTime(float time) { time_=time; }
-  const DetId& detid() const { return id_; }
-  uint32_t flags() const { return flags_; }
-  void setFlags(uint32_t flags) { flags_=flags; }
-  void setFlagField(uint32_t value, int base, int width=1);
-  uint32_t flagField(int base, int width=1) const;
-  void setAux(uint32_t value) { aux_=value; }
-  uint32_t aux() const { return aux_; }
+  constexpr float energy() const { return energy_; }
+  constexpr void setEnergy(float energy) { energy_=energy; }
+  constexpr float time() const { return time_; }
+  constexpr void setTime(float time) { time_=time; }
+  constexpr const DetId& detid() const { return id_; }
+  constexpr uint32_t flags() const { return flags_; }
+  constexpr void setFlags(uint32_t flags) { flags_=flags; }
+  constexpr void setFlagField(uint32_t value, int base, int width=1) {
+#ifdef __CUDA_ARCH__
```

Comment:
are you sure we need these `#ifdef  __CUDA_ARCH__` blocks ? 
`std::min` and `std::max` should be `constexpr`, so they should be available in device code as well.

Reply:
true, removed